src/backend/distributed/commands/vacuum.c

```diff
@@ -22,9 +22,162 @@
 #include "utils/builtins.h"
 #include "utils/lsyscache.h"
 
+#if PG_VERSION_NUM >= 120000
+
+#include "commands/vacuum.h"
+#include "commands/defrem.h"
+
+/*
+ * This is mostly ExecVacuum from Postgres's commands/vacuum.c
+ */
+
+/*
+ * A wrapper function of defGetBoolean().
+ *
+ * This function returns VACOPT_TERNARY_ENABLED and VACOPT_TERNARY_DISABLED
+ * instead of true and false.
+ */
+static VacOptTernaryValue
+get_vacopt_ternary_value(DefElem *def)
+{
+	return defGetBoolean(def) ? VACOPT_TERNARY_ENABLED : VACOPT_TERNARY_DISABLED;
+}
+
+
+static int
+VacuumStmt_options(VacuumStmt *vacstmt)
```

Comment:
The order we follow is usually declarations -> definitions, and we have forward declarations for every function. In older times that we were stricter in style, the definitions followed their use. That is `VacuumStmt_options()`, etc. should be defined after the place it is used. The idea being that when we read the file from top to down, we first read the higher level concepts firsts and then continue to the lower level details.